### ISSUE TITLE
Migration scripts not found when using Azure App Service

### ISSUE THREAD

**Issue by ld0614** [lecaillon/Evolve#76]
Configuration:
C# .Net 4.6.2
Azure SQL Server V12
Azure App Service

Migration Launcher:
```
try
{
	string connectionString = Settings.GetSettings().DatabaseConnectionString;
	SqlConnection cnx = new SqlConnection(connectionString);
	Evolve.Evolve evolve = new Evolve.Evolve(cnx, msg => Trace.TraceInformation(msg))
	{
		Driver = "SqlClient", //Connect to SQL Server
		Locations = new List<string> { "db/migrations" },
		IsEraseDisabled = true, //Do not allow Erasing of schema
		EnableClusterMode = true //Ensure that no other instances also attempt to migrate the database as the same time
	};

	evolve.Migrate();
}
catch (Exception ex)
{
	Trace.TraceError("Database migration failed.", ex);
	throw;
}
}
```
SQL files have been written and are uploaded to the web server:

![image](https://user-images.githubusercontent.com/20439950/44859804-b81aa700-ac6c-11e8-997e-7dcc9ebf342e.png)

During start the following logs are provided
```
Application: 2018-08-30T14:46:57  PID[5928] Information Executing Migrate...
Application: 2018-08-30T14:46:57  PID[5928] Information Evolve initialized.
Application: 2018-08-30T14:46:57  PID[5928] Information No metadata found.
Application: 2018-08-30T14:46:57  PID[5928] Information Database is up to date. No migration needed.
```

And a single blank changelog table has been created:
![image](https://user-images.githubusercontent.com/20439950/44859922-03cd5080-ac6d-11e8-9b50-8d55721a91d3.png)

As far as I can tell I've followed all the requirements for using Evolve but I'm unsure of how to troubleshoot this issue further, any assistance would be greatly appreciated.

**Comment by lecaillon** [comment]
Hi @ld0614 

Since the `changelog` table has been created, it means that Evolve is able to connect to your database.
Your migration scripts seem to be well named.
So I think it does not find any of your sql migration files.
Have you set the property `Copy to output directory` to `Copy always` on each of your sql file ?

**Comment by ld0614** [comment]
Hi @lecaillon, thanks for getting back so soon, the screenshot above is a view of the web server file structure showing the files on the web server.  The below screenshot if the file configuration.
![image](https://user-images.githubusercontent.com/20439950/44861539-a1764f00-ac70-11e8-95ab-87418cd6c15d.png)

The only thing I really stray from your example is in the code launch, in Global.asax.cs I have the following line:
```
UpdateDatabase.UpdateDatabaseUsingEvolve();
```

This then calls the updateDatabase class which is saved in App_Start\UpdateDatabase.cs file.  The db\migration path is at the root of the directory.

I have also attempted to place the files at /bin/App_Start/db/migrations as the dlls for the website are at /bin.  After updating the file locations to either bin/App_Start/db/migrations or Add_Start/db/migrations nither returned a different log message.

**Comment by lecaillon** [comment]
ok, in fact Evolve searches migration files in the current app directory combine with the value of the `Location` parameter. Or in the `Location` parameter only if you have define a full path.
So in our case I see the `db` folder is next the `bin` folder.
Thus I would try:
```
Locations = new List { "../db/migrations" },
```

**Comment by ld0614** [comment]
Thanks for all your assistance, after a few experiments I discovered that the working directory on an Azure App Service is fixed to D:\Windows\System32.  thus all uses of DirectoryInfo where branching off of there.  using ../db/migrations" didn't help either as this returned to the root of the D drive.  To fix I added the following code:
```
string currentPath = System.Web.HttpContext.Current.Server.MapPath("~");
Locations = new List<string> { currentPath + "db\\migrations" },
```

This maps to the correct root folder which can then be changed to the correct relative folder.

To assist with future debugging may I suggest a branch from here: https://github.com/lecaillon/Evolve/blob/25dab78a7dd969ef4316063aa43b2fa1ed990329/src/Evolve/Migration/FileMigrationLoader.cs#L29 which would write a message if it was unable to locate the directory (ideally with the offending path)?
